modules/userId/index.js

```diff
@@ -465,13 +465,103 @@ function getUserIdsAsEids() {
   return createEidsArray(getCombinedSubmoduleIds(initializedSubmodules));
 }
 
+/**
+* This function will be exposed in the global-name-space so that userIds can be refreshed after initialization.
+* @param {String[]} submoduleNames submodules to be refreshed
+*/
+function refreshUserIds(submoduleNames) {
+  if (!utils.isArray(submoduleNames)) {
+    return;
+  }
+
+  if (!initializedSubmodules) {
+    utils.logWarn(`${MODULE_NAME} - refresh called before submodule initialization`);
```

Comment:
I would have expected this method to initialize the submodule if it has not already been initialized. Otherwise you might end up with race conditions  where you need to wait for a first bid to occur before you can call this method.

So I would expect this not to return early here. Instead of iterating over `initializedSubmodules` below, I would iterate over `userIdModules` and filter by the name.